### PR TITLE
implement ewmh atom _NET_FRAME_EXTENTS

### PR DESCRIPTION
**Pull request by codic12**
Also changed a hardcoded zero to XCB_WINDOW_NONE just cuz.

### CODE REVIEW

File: src/ewmh/ewmh.cpp [hunk 1]
```diff
@@ -52,4 +52,9 @@ void EWMH::updateClientList() {
 
     xcb_change_property(g_pWindowManager->DisplayConnection, XCB_PROP_MODE_REPLACE, g_pWindowManager->Screen->root, HYPRATOMS["_NET_CLIENT_LIST"], XCB_ATOM_WINDOW,
         32, windowsList.size(), ArrWindowList);
-}
\ No newline at end of file
+}
+
+void EWMH::setFrameExtents(xcb_window_t w) {
+	uint32_t extents[4] = {ConfigManager::getInt("border_size"), ConfigManager::getInt("border_size"), ConfigManager::getInt("border_size"), ConfigManager::getInt("border_size")};
```
Comment (vaxerski) [inline src/ewmh/ewmh.cpp:58]: This is slow. You ale calling the getInt method that can take ~7μs 4 times. Maybe call it once?

```
const auto BORDERSIZE = ConfigManager::getInt("border_size");
uint32_t extents = {BORDERSIZE,BORDERSIZE,BORDERSIZE,BORDERSIZE};
```

File: src/events/events.cpp [hunk 2]
```diff
@@ -420,6 +420,7 @@ void Events::eventMapWindow(xcb_generic_event_t* event) {
 
     // EWMH
     EWMH::updateClientList();
+    EWMH::setFrameExtents(E->window);
```
Comment (vaxerski) [inline src/events/events.cpp:423]: Incorrect. Will not update on config reload.

**Review by vaxerski — CHANGES_REQUESTED** [review]
Left a review. 1 bug, 1 optimization.

**Comment by codic12** [comment]
I've fixed both issues in the review. The border size is cached and used all 4 times in the array declaration for extents. The border size is also checked after config is reloaded, and if different it will call an EWMH::refreshAllExtents function.

**Pull request merged**